universe/auto_syncer.go

```diff
@@ -624,6 +563,74 @@ func (f *FederationEnvoy) handleTickEvent() error {
 	return nil
 }
 
+// handlePushRequest is called each time a new push request is received. It will
+// perform an asynchronous registration with the local Universe registrar, then
+// push the proof leaf out in an async manner to the federation members.
+func (f *FederationEnvoy) handlePushRequest(pushReq *FederationPushReq) error {
+	if pushReq == nil {
+		return fmt.Errorf("nil push request")
+	}
+
+	// First, we'll attempt to registrar the proof leaf with the local
+	// registrar server.
+	ctx, cancel := f.WithCtxQuit()
+	newProof, err := f.cfg.LocalRegistrar.UpsertProofLeaf(
+		ctx, pushReq.ID, pushReq.Key, pushReq.Leaf,
+	)
+	cancel()
```

Comment:
nit: we can use `defer` to cancel these contexts, since we're in a method now.

Reply:
I've added fixups which use `defer`. Please let me know if that's what you intended.